### PR TITLE
[static setitem] Support the index is Tensor, step>1 and step<0

### PR DESCRIPTION
**Pull request by liym27**
### PR types

New features 
### PR changes

APIs
### Describe

功能支持：静态图支持 slice 赋值，inplace操作。

本 PR 支持场景：
key支持：反向步长，即 step 小于 0
value支持：int, float, numpy.ndarray, Paddle Tensor
dtype: int32, int64, float32, float64, bool
其他：默认支持广播

```
tensor_x = paddle.to_tensor(np.zeros((5)).astype(np.float32)) # [0,0,0,0,0]
tensor_x[2::-1] = np.array([1,2,3]) # tensor_x : [3,2,1,0,0]
tensor_x[3:0:-2] = paddle.ones([1]) # tensor_x : [0,1,0,1,0]
```

---

**当前 setitem 功能，索引类型支持情况如下**

索引类型 | 示例 | 动态图 | 静态图
-- | -- | -- | --
标量 | x[1]   = y | √ | √
切片 | x[1:2]   = y | √ | √
跨步长 | x[1:10:3]   = y | √ | https://github.com/PaddlePaddle/Paddle/pull/30888
反向步长 | x[1:10:-2]   = y | √ | https://github.com/PaddlePaddle/Paddle/pull/30897
省略[Ellipsis] | x[…,   0] = y | √ | https://github.com/PaddlePaddle/Paddle/pull/30836
Tensor | x[tensor_i]   = y | × | **本 PR 支持**

### CODE REVIEW

File: python/paddle/fluid/utils.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (zhhsplendid) [inline python/paddle/fluid/utils.py]: 2018？

File: python/paddle/fluid/utils.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import six
+from .framework import Variable
+
+
+def _contain_var(list_or_tuple):
```
Comment (zhhsplendid) [inline python/paddle/fluid/utils.py]: I would suggest not to create new `utils.py` file for managing paddle modules.

Can we put them in "paddle/fluid/layers/utils.py" ?

Comment (liym27) [reply python/paddle/fluid/utils.py]: Thanks,done

File: python/paddle/fluid/utils.py (hunk 1 above)
Comment (liym27) [reply python/paddle/fluid/utils.py]: Removed this file.

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
LGTM for framework.py

**Pull request merged**